src/Agent.Sdk/SecretMasking/OssSecretMasker.cs

```diff
@@ -85,7 +122,109 @@ public void Dispose()
 
     public string MaskSecrets(string input)
     {
-        return _secretMasker.MaskSecrets(input);
+        if (TelemetryEnabled)
+        {
+            Interlocked.Add(ref _charsScanned, input.Length);
+            Interlocked.Increment(ref _stringsScanned);
+        }
+
+        return _secretMasker.MaskSecrets(input, _detectionAction);
+    }
+
+    public bool TelemetryEnabled
+    {
+        get => _detectionAction != null;
+        set
+        {
+            if (value != TelemetryEnabled)
+            {
+                if (value)
+                {
+                    _detectionTelemetry = new ConcurrentDictionary<string, string>();
+                    _detectionAction = this.ProcessDetectionForTelemetry;
+                }
+                else
+                {
+                    throw new InvalidOperationException("SecretMasker telemetry cannot be turned off if it has been turned on before.");
```

Comment:
hope we can refactor the exception out.  we'll 'never' hit it but it would be nice to be safe then sorry

Reply:
Done. It is now part of the static API that you can only flip it on via EnableTelemetry() and there is no exceptional case.